src/node_http_parser.cc

```diff
@@ -759,7 +759,9 @@ void InitHttpParser(Local<Object> target,
     methods->Set(num, FIXED_ONE_BYTE_STRING(env->isolate(), #string));
   HTTP_METHOD_MAP(V)
 #undef V
-  t->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "methods"), methods);
+
+  auto methods_string = FIXED_ONE_BYTE_STRING(env->isolate(), "methods");
+  t->SetNativeDataProperty(methods_string, DataGetter, nullptr, methods);
```

Comment:
this one is a bit tricker. You'll have to create the array in a getter. Here's some pseudo code for the getter

Local<Private> key = Private::ForApi("http-methods");
if (info.Holder()->HasPrivate(key)) {
  info.GetReturnValue.Set(info.Holder()->GetPrivate(key));
  return;
}
Local<Array> methods = Array::New(info.GetIsolate()->GetCurrent()); // this is the key - the array is now in the correct context, and not shared with others.
// ... populate array
info.Holder()->SetPrivate(key, methods);
info.GetReturnValue().Set(methods);
return;
}